docs/src/token-2022/extensions.md

```diff
@@ -2,4 +2,997 @@
+to fit room for more extensions.
+
+#### Example: Reallocating existing account to enable required memo transfers
+
+<Tabs className="unique-tabs" groupId="language-selection">
+  <TabItem value="cli" label="CLI" default>
+
+CLI support coming soon!
+
+  </TabItem>
+  <TabItem value="jsx" label="JS">
+
+```jsx
+import {
+    clusterApiUrl,
+    sendAndConfirmTransaction,
+    Connection,
+    Keypair,
+    Transaction,
+    LAMPORTS_PER_SOL,
+} from '@solana/web3.js';
+import {
+    createAccount,
+    createMint,
+    createEnableRequiredMemoTransfersInstruction,
+    createReallocateInstruction,
+    ExtensionType,
+    TOKEN_2022_PROGRAM_ID,
+} from '@solana/spl-token';
+
+(async () => {
+    const connection = new Connection(clusterApiUrl('devnet'), 'confirmed');
+
+    const payer = Keypair.generate();
+    const airdropSignature = await connection.requestAirdrop(payer.publicKey, 2 * LAMPORTS_PER_SOL);
+    await connection.confirmTransaction(airdropSignature);
+
+    const mintAuthority = Keypair.generate();
+    const decimals = 9;
+    const mint = await createMint(
+        connection,
+        payer,
+        mintAuthority.publicKey,
+        mintAuthority.publicKey,
+        decimals,
+        undefined,
+        undefined,
+        TOKEN_2022_PROGRAM_ID
+    );
+
+    const owner = Keypair.generate();
+    const account = await createAccount(
+        connection,
+        payer,
+        mint,
+        owner.publicKey,
+        undefined,
+        undefined,
+        TOKEN_2022_PROGRAM_ID
+    );
+
+    const extensions = [ExtensionType.MemoTransfer];
+    const transaction = new Transaction().add(
+        createReallocateInstruction(
+            account,
+            payer.publicKey,
+            extensions,
+            owner.publicKey,
+            undefined,
+            TOKEN_2022_PROGRAM_ID
+        ),
+        createEnableRequiredMemoTransfersInstruction(account, owner.publicKey, [], TOKEN_2022_PROGRAM_ID)
+    );
+    await sendAndConfirmTransaction(connection, transaction, [payer, owner], undefined);
+})();
+```
+
+  </TabItem>
+</Tabs>
+
+### Interest-Bearing Tokens
+
+Tokens that constantly grow or decrease in value have many uses in the real world.
+The most well known example is a bond.
+
+With Token, this has only been possible through proxy contracts that require
+regular rebase or update operations.
+
+With the Token-2022 extension model, however, we have the possibility to change
+how the UI amount of tokens are represented. Using the `InterestBearingMint`
+extension and the `amount_to_ui_amount` instruction, you can set an interest
+rate on your token and fetch its amount with interest at any time.
+
+Interest is continuously compounded based on the timestamp in the network. Due
+to drift that may occur in the network timestamp, the accumulated interest could
+be lower than the expected value. Thankfully, this is rare.
+
+Note that no new tokens are ever created, the UI amount returns the amount of tokens
```

Comment:
```suggestion
**Note**: No new tokens are ever created, the UI amount returns the amount of tokens
```

Reply:
Also, does this mean that there needs to be a separate program that creates the tokens based on the interest accumulated? This statement seems like this feature is just cosmetic for UI vs actually creating new tokens as they are held.

Reply:
The feature is in fact entirely cosmetic.  I'll make that clearer